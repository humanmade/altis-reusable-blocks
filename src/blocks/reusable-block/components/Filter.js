import PropTypes from 'prop-types';

import {
	SelectControl,
	TextControl,
} from '@wordpress/components';
import { __ } from '@wordpress/i18n';

const Filter = ( {
	categoriesList,
	searchCategory,
	searchKeyword,
	updateSearchCategory,
	updateSearchKeyword
} ) => {

	const categoriesListOptions = categoriesList
		? categoriesList.map( ( { id, name } ) => ( {
			label: name,
			value: id,
		} ) )
		: [];

	if ( categoriesListOptions.length ) {
		categoriesListOptions.unshift( { label: '-- Select Category --', value: '' } );
	}

	return(
		<div className="block-editor-reusable-blocks-inserter__filter">
			<TextControl
				className="block-editor-reusable-blocks-inserter__filter-search-keyword"
				label={ __( 'Search: ', 'enhanced-reusable-blocks' ) }
				onChange={ updateSearchKeyword }
				value={ searchKeyword }
			/>
			<SelectControl
				className="block-editor-reusable-blocks-inserter__filter-search-category"
				label={ __( 'Category: ', 'enhanced-reusable-blocks' ) }
				onChange={ updateSearchCategory }
				options={ categoriesListOptions }
				value={ searchCategory }
			/>
		</div>
	);
};

Filter.propTypes = {
	categoriesList: PropTypes.array.isRequired,
	searchCategory: PropTypes.number,
	searchKeyword: PropTypes.string,
	updateSearchCategory: PropTypes.func.isRequired,
	updateSearchKeyword: PropTypes.func.isRequired,
};

export default Filter;
