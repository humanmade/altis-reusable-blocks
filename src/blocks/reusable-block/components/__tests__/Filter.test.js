import { shallow } from 'enzyme';

import Filter from '../Filter';

describe( '<Filter />', () => {

	test( 'renders as expected if no categories', () => {
		const result = shallow(
			<Filter
				categoriesList={ [] }
				searchID={ 42 }
				updateSearchCategory={ () => {} }
				updateSearchKeyword={ () => {} }
			/>
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders as expected with categories', () => {
		const categoriesList = [
			{
				id: 1,
				name: 'First',
			},
			{
				id: 2,
				name: 'Second',
			},
			{
				id: 3,
				name: 'Third',
			},
		];

		const result = shallow(
			<Filter
				categoriesList={ categoriesList }
				updateSearchCategory={ () => {} }
				updateSearchKeyword={ () => {} }
			/>
		);

		expect( result ).toMatchSnapshot();
	} );

} );
