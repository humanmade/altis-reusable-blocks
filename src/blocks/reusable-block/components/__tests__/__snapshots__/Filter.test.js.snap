// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders as expected if no categories 1`] = `
<div
  className="block-editor-reusable-blocks-inserter__filter"
>
  <TextControl
    className="block-editor-reusable-blocks-inserter__filter-search-keyword"
    label="Search: "
    onChange={[Function]}
    placeholder="Search by keywords, post ID, or block ID..."
  />
  <SelectControl
    className="block-editor-reusable-blocks-inserter__filter-search-category"
    disabled={42}
    label="Category: "
    onChange={[Function]}
    options={Array []}
  />
</div>
`;

exports[`<Filter /> renders as expected with categories 1`] = `
<div
  className="block-editor-reusable-blocks-inserter__filter"
>
  <TextControl
    className="block-editor-reusable-blocks-inserter__filter-search-keyword"
    label="Search: "
    onChange={[Function]}
    placeholder="Search by keywords, post ID, or block ID..."
  />
  <SelectControl
    className="block-editor-reusable-blocks-inserter__filter-search-category"
    label="Category: "
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "-- Select Category --",
          "value": "",
        },
        Object {
          "label": "First",
          "value": 1,
        },
        Object {
          "label": "Second",
          "value": 2,
        },
        Object {
          "label": "Third",
          "value": 3,
        },
      ]
    }
  />
</div>
`;
